File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Integer[] happi= new Integer[happiness.length];
        for(int i=0;i<happiness.length;i++)
            happi[i]=happiness[i];
        Arrays.sort(happi, Collections.reverseOrder());
        long ans=0;
        for(int i=0;i<k;i++){
            ans += Math.max(happi[i]-i,0); 
        }
        return ans;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        N = len(happiness)
        
        happiness.sort(reverse= True)
        ans = 0
        for i,h in enumerate(happiness[:k]):
            ans += max((h-i),0)
        
        return ans
        